internal/backend/remote-state/gcs/backend_test.go

```diff
@@ -43,40 +43,115 @@ const (
 
 var keyRingLocation = os.Getenv("GOOGLE_REGION")
 
-func TestStateFile(t *testing.T) {
-	t.Parallel()
+func preCheckTestAcc(t *testing.T) {
+	if v := os.Getenv("TF_ACC"); v == "" {
+		t.Skip("TF_ACC must be set to run acceptance tests, as they provision real resources")
+	}
+}
 
-	cases := []struct {
-		prefix        string
-		name          string
-		wantStateFile string
-		wantLockFile  string
-	}{
-		{"state", "default", "state/default.tfstate", "state/default.tflock"},
-		{"state", "test", "state/test.tfstate", "state/test.tflock"},
-		{"state", "test", "state/test.tfstate", "state/test.tflock"},
-		{"state", "test", "state/test.tfstate", "state/test.tflock"},
-	}
-	for _, c := range cases {
-		b := &Backend{
-			prefix: c.prefix,
-		}
+func preCheckEnvironmentVariables(t *testing.T) {
 
-		if got := b.stateFile(c.name); got != c.wantStateFile {
-			t.Errorf("stateFile(%q) = %q, want %q", c.name, got, c.wantStateFile)
-		}
+	// TODO - implement a separate function specific to credentials which will iterate through a list of ENV names and return first value found.
```

Comment:
Is this a note to self to be answered in this PR, or for later?

Reply:
Something for later - this code would benefit from a lot of refactoring but I didn't want to do it all in a single PR!